### PR TITLE
Internal Gallery Extension

### PR DESCRIPTION
**Pull request by city-unit**
This is a quick and dirty gallery feature powered by [nanogallery2 ](https://nanogallery2.nanostudio.org/)(open source, gpl, etc.)
It reveals all of the image files found in the current characters folder. This could easily be expanded to all files in the user data folder if desired. Currently the menu is under the character dropdown menu. 
Some notes:
- I had to add a new endpoint on the server side to reveal image in a folder since we do not allow folder enumeration.
- When the gallery is called up, it does un-set the body from being `position: fixed` if it is currently the case (and resets it when finished) because this breaks the gallery (and basically all galleries that I tested, this cost me a long time). 
- I added an event emitter to the dropdown switch statement because extension cannot catch the state of the dropdown before it is reset to nothing by the line directly after the switch statement. 
- I put the gallery in a standard popup because it keeps the correct style and whatnot. It's not like the ideal container (and I think there are ways to make it bigger that I haven't tried), but it's consistent and works for this pretty well. 

If any of this is objectionable, let me know, I'm happy to try to work around it. 
I've tested it some, it works well on desktop and mobile. It's not the lightest of the libraries in the world 250kb I think, but it's featureful and well supported (but there are alternatives). Take it for a spin and let me know what you think.  

(sorry, I forgot to switch feature branches)

### CODE REVIEW

File: public/scripts/extensions/gallery/index.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+let firstTime = true;
+
+/**
+ * Loads either a CSS or JS file and appends it to the appropriate document section.
+ * 
+ * @param {string} url - The URL of the file to be loaded.
+ * @param {string} type - The type of file to load: "css" or "js".
+ * @returns {Promise} - Resolves when the file has loaded, rejects if there's an error or invalid type.
+ */
+function loadFile(url, type) {
```
Comment (Cohee1207) [inline public/scripts/extensions/gallery/index.js]: I think we could add common utility functions for extensions to extensions.js or utils.js
I've seen the same carbon copy in the timelines (with an interesting comment 🤣 )

Comment (Cohee1207) [reply public/scripts/extensions/gallery/index.js]: I also considered allowing extensions to load arbitrary assets from their folders into the in-memory array of buffers. Kind of like what Coqui TTS would want to load a JSON file to get rid of a hard-coded path to JSON. 
`getAsset(extensionName, assetId) => Buffer`

File: server.js [hunk 2]
```diff
@@ -2679,6 +2679,18 @@ app.post('/uploadimage', jsonParser, async (request, response) => {
...
 });
 
+app.get('/listimgfiles/:folder', (req, res) => {
+    const directoryPath = path.join(__dirname, 'public/user/images/', req.params.folder);
+    console.log(directoryPath);
+    fs.readdir(directoryPath, (err, files) => {
+        if (err) {
+            return res.status(500).send({ error: "Unable to retrieve files" });
+        }
+        const imgFiles = files.filter(file => ['.jpg', '.jpeg', '.png', '.gif'].includes(path.extname(file)));
```
Comment (Cohee1207) [inline server.js]: Maybe it's better to filter by "image/*" mime type instead? Cause, hmm, .webp exists

Comment (city-unit) [reply server.js]: Easy enough, we can just not filter for now.

File: public/scripts/extensions/gallery/index.js (hunk 1 above)
Comment (city-unit) [reply public/scripts/extensions/gallery/index.js]: Hmm yeah that could be nice. I've moved it to utils for now. We can make that better or move it elsewhere in the future.

**Pull request merged**